### PR TITLE
Add Heroku add-on

### PR DESCRIPTION
**Pull request by hoangmirs**
Solve #52 

## What happened 👀

- Add Heroku addon
- Add Heroku with terraform files and directory
- Update template README and .gitignore
 
## Insight 📝

N/A
 
## Proof Of Work 📹

- Generate the project
Screen Shot 2564-07-16 at 19 02 18

- Unit test is passed
Screen Shot 2564-07-16 at 19 03 28

### CODE REVIEW

File: hooks/post_gen_project.py [hunk 1]
```diff
@@ -17,15 +17,47 @@ def print_log(message):
...
+    """
+    shutil.rmtree(os.path.join(
+        PROJECT_DIRECTORY, "deploy/heroku"
+    ))
+    os.remove(os.path.join(
+        PROJECT_DIRECTORY, ".github/workflows/deploy.yml"
+    ))
+
+
+def remove_empty_folders():
```
Comment (gutakk) [inline hooks/post_gen_project.py:32]: @hoangmirs Why do we need this function?

Comment (hoangmirs) [reply hooks/post_gen_project.py:32]: In `remove_heroku_files` function, I just remove `deploy/heroku`, so it still has the empty `deploy` folder.
I had planned to remove the entire `deploy` folder, but I think we might have another one in that folder, so I went with this way

**Comment by malparty** [comment]
@gutakk  @hoangmirs 
Just as a cross-check, I'm trying to deploy to Heroku and at the `terraform plan` cmd, I got a small warning (and a note):
```
Plan: 4 to add, 0 to change, 0 to destroy.
╷
│ Warning: Value for undeclared variable
│
│ The root module does not declare a variable named "redis_plan" but a value was found in file "terraform.tfvars". If you meant to use this value, add a "variable" block to the
│ configuration.
│
│ To silence these warnings, use TF_VAR_... environment variables to provide certain "global" settings to all configurations in your organization. To reduce the verbosity of
│ these warnings, use the -compact-warnings option.
╵

Note: You didn't use the -out option to save this plan, so Terraform can't guarantee to take exactly these actions if you run "terraform apply" now.
```

It does not seems to be blocker, but I'm still raising it in case it's not showing up in your tests. ;-)


Note that I cannot test the next cmd (terraform apply) as I do not have access to any Heroku organization~

**Comment by hoangmirs** [comment]
> @gutakk @hoangmirs
> Just as a cross-check, I'm trying to deploy to Heroku and at the `terraform plan` cmd, I got a small warning (and a note):
> 
> It does not seems to be blocker, but I'm still raising it in case it's not showing up in your tests. ;-)
> 
> Note that I cannot test the next cmd (terraform apply) as I do not have access to any Heroku organization~

@malparty Thanks for pointing it out. I removed the redundant redis variable in https://github.com/nimblehq/gin-templates/pull/55/commits/6bb3e3bc58650350ab29191f688385d6536f0579
@gutakk Please fetch this branch again if you wanna add more content

**Review by gutakk — APPROVED** [review]

**Comment by gutakk** [comment]
> > @gutakk @hoangmirs
> > Just as a cross-check, I'm trying to deploy to Heroku and at the `terraform plan` cmd, I got a small warning (and a note):
> > It does not seems to be blocker, but I'm still raising it in case it's not showing up in your tests. ;-)
> > Note that I cannot test the next cmd (terraform apply) as I do not have access to any Heroku organization~
> 
> @malparty Thanks for pointing it out. I removed the redundant redis variable in [6bb3e3b](https://github.com/nimblehq/gin-templates/commit/6bb3e3bc58650350ab29191f688385d6536f0579)
> @gutakk Please fetch this branch again if you wanna add more content

LGTM

**Review by malparty — APPROVED** [review]

**Pull request merged**